plugins/auth-backend/src/providers/gitlab/provider.ts

```diff
@@ -78,6 +78,14 @@ export class GitlabAuthProvider implements OAuthProviderHandlers {
       idToken: params.id_token,
     };
 
+    // gitlab provides an id numeric value (123)
+    // as a fallback
+    let id = passportProfile!.id;
+
+    if (profile.email) {
+      id = profile.email.split('@')[0];
```

Comment:
@Rugvip using this to set the id, below we return the providerInfo, profile and backstageIdentity with the id

Reply:
Yep, looks good! It's really the best we can do for now before we have a clear concept of identities.

Reply:
what happens when 2 people have same first part of email (x@domain1 and x@domain2) - this seems risky/should be using fully qualified email if there is no other unique id?

Reply:
Stuff breaks. This is waiting for something like https://github.com/spotify/backstage/issues/1401, where it's possible to go an look up an actual identity via an external email or id. For now this is just a placeholder to return some kind of meaningful ID, but it doesn't actually do anything yet.

Reply:
how is removing the domain (without checking it being maybe a base one) helpful though?